### PR TITLE
[Discuss] Update Codeowners file

### PR DESCRIPTION
**Pull request by ruflin**
The following is a detailed update of the codeowners file for the different teams. Each top level directory is listed with the team that owns it. In some cases like modules more specific details were added. Every time we see some specific directories belong to a team, the Codeowners file should be updated.

The goal of this PR is to have alignment across all the teams on ownership.

Few additional notes:

* The robots team is not an owner of any files / directories. This is by design as the final ownership of the build is with the data plane team.
* Modules are owned by default by the integrations team. More details must be added here.
* system tests are not covered in details

### CODE REVIEW

File: .github/CODEOWNERS
Comment (ruflin) [inline .github/CODEOWNERS:21]: Does the data plan team have to review all changes to k8s for example?

File: .github/CODEOWNERS
Comment (ruflin) [inline .github/CODEOWNERS:37]: @masci Current most modules or owned by integrations or security team is my understanding? Does it make sense that the default ownership is with integrations?

File: .github/CODEOWNERS
Comment (ruflin) [inline .github/CODEOWNERS:70]: @masci Same as above.

File: .github/CODEOWNERS
Comment (ruflin) [inline .github/CODEOWNERS:81]: @andrewvc This kind of looks off on this level.

File: .github/CODEOWNERS
Comment (jlind23) [reply .github/CODEOWNERS:21]: I do not think so.

**Review by ph — COMMENTED** [review]
We might want to add `libbeat/management` to the control plane. This PR will go in 7.x too right?

**Comment by ruflin** [comment]
Good question from @ph if this should be backported? So far I was thinking no as the changes to 7.x are hopefully minimal. I would go so far that all changes going into 7.x must be reviewed by the control and data plane team from now on.

I'll add the libbeat/management change.

**Comment by jlind23** [comment]
I would vote for a backport as it won't have a huge impact and may avoid potential "silent" merge 👍🏼

File: .github/CODEOWNERS
Comment (cmacknz) [reply .github/CODEOWNERS:21]: I don't mind the data plane team being listed as an owner here. SDHs for filebeat/metricbeat on k8s will probably end up with the data plane team, we should know what the deployment set up is.

We could list two codeowners here, one for the beat and one for whoever maintains the deployment types.

File: .github/CODEOWNERS
Comment (cmacknz) [inline .github/CODEOWNERS:36]:
```suggestion
# Modules are by default owned by the integrations team
```

File: .github/CODEOWNERS
Comment (cmacknz) [inline .github/CODEOWNERS:55]:
```suggestion
# Journalbeat is owned by the data plane team
```

File: .github/CODEOWNERS
Comment (cmacknz) [reply .github/CODEOWNERS:71]: The system module should be the data plane team https://github.com/elastic/beats/tree/main/metricbeat/module/system

File: .github/CODEOWNERS
Comment (ruflin) [reply .github/CODEOWNERS:21]: @cmacknz Do you mean for `/deploy/kubernetes` we add data plane team and the team from @ChrsMark ? I like this idea because it ensures any change to k8s has the eyes from both teams on it.

File: .github/CODEOWNERS
Comment (ruflin) [reply .github/CODEOWNERS:73]: Added the system module to the data plane team.

File: .github/CODEOWNERS
Comment (cmacknz) [reply .github/CODEOWNERS:21]: Yes, let's add both teams.

**Review by cmacknz — APPROVED** [review]

**Review by ph — APPROVED** [review]

File: .github/CODEOWNERS
Comment (andrewvc) [reply .github/CODEOWNERS:52]: yep!

File: .github/CODEOWNERS
Comment (andrewvc) [reply .github/CODEOWNERS:87]: That's not right, we can remove this. That directory only exists in `heartbeat` and `x-pack/heartbeat` which should already be covered.

Comment (ruflin) [reply .github/CODEOWNERS:87]: Can you follow up with a PR removing the directory?

File: .github/CODEOWNERS
Comment (ruflin) [reply .github/CODEOWNERS:37]: Happy to add a list with the detailed owners. @andrewkroh Are you ok with being the default owner?

**Comment by ruflin** [comment]
Thanks for the list @andrewkroh , updated the change with it.

**Comment by andrewkroh** [comment]
If this is backported will it affect the ability to merge backport PRs without review? Today, we don't require any review of backports.

File: .github/CODEOWNERS
Comment (andrewkroh) [reply .github/CODEOWNERS:37]: I think we should do the effort now of defining the module owners. It will reduce noise on the teams and prevent any ambiguity of responsibility.

**Review by jlind23 — APPROVED** [review]

**Comment by ruflin** [comment]
@v1v Will this have affects on the backporting tool?

**Comment by v1v** [comment]
> @v1v Will this have affects on the backporting tool?

I believe so, it's a native support from GitHub, see https://github.com/Mergifyio/mergify-engine/issues/464

> If this is backported will it affect the ability to merge backport PRs without review? Today, we don't require any review of backports.

IIUC, that's not the case with the existing branch protection:

image

That's the one for the [`8.*`](https://github.com/elastic/beats/settings/branch_protection_rules/22874483) branches

**Comment by ruflin** [comment]
Based on the comment from @v1v it seems, we don't require reviews from Code owners on "backport" branches. This is great for getting this change in but TBH it makes a bit nervous as it means in release branches changes could be merged without a code owner review. As elasticmachine can merge without reviews, I guess we could still turn this on and the automatic backporting would keep working?

File: .github/CODEOWNERS
Comment (ruflin) [reply .github/CODEOWNERS:37]: @masci @andrewkroh Could you put the list together?

Comment (andrewkroh) [reply .github/CODEOWNERS:37]: Derived from the elastic/integrations CODEOWNERS and git history.

```
/filebeat/module/apache @elastic/integrations
/filebeat/module/auditd @elastic/security-external-integrations
/filebeat/module/elasticsearch @elastic/integrations
/filebeat/module/haproxy @elastic/integrations
/filebeat/module/icinga @elastic/integrations
/filebeat/module/iis @elastic/integrations
/filebeat/module/kafka @elastic/integrations
/filebeat/module/kibana @elastic/integrations
/filebeat/module/logstash @elastic/integrations
/filebeat/module/mongodb @elastic/integrations
/filebeat/module/mysql @elastic/security-external-integrations
/filebeat/module/nats @elastic/integrations
/filebeat/module/nginx @elastic/integrations
/filebeat/module/osquery @elastic/security-asset-management
/filebeat/module/pensando @elastic/security-external-integrations
/filebeat/module/postgresql @elastic/integrations
/filebeat/module/redis @elastic/integrations
/filebeat/module/santa @elastic/security-external-integrations
/filebeat/module/system @elastic/ecosystem
/filebeat/module/traefik @elastic/integrations
/x-pack/filebeat/module/activemq @elastic/integrations
/x-pack/filebeat/module/aws @elastic/obs-cloud-monitoring
/x-pack/filebeat/module/awsfargate @elastic/obs-cloud-monitoring
/x-pack/filebeat/module/azure @elastic/obs-cloud-monitoring
/x-pack/filebeat/module/barracuda @elastic/security-external-integrations
/x-pack/filebeat/module/bluecoat @elastic/security-external-integrations
/x-pack/filebeat/module/cef @elastic/security-external-integrations
/x-pack/filebeat/module/checkpoint @elastic/security-external-integrations
/x-pack/filebeat/module/cisco @elastic/security-external-integrations
/x-pack/filebeat/module/coredns @elastic/security-external-integrations
/x-pack/filebeat/module/crowdstrike @elastic/security-external-integrations
/x-pack/filebeat/module/cyberarkpas @elastic/security-external-integrations
/x-pack/filebeat/module/cylance @elastic/security-external-integrations
/x-pack/filebeat/module/envoyproxy @elastic/security-external-integrations
/x-pack/filebeat/module/f5 @elastic/security-external-integrations
/x-pack/filebeat/module/fortinet @elastic/security-external-integrations
/x-pack/filebeat/module/gcp @elastic/security-external-integrations
/x-pack/filebeat/module/google_workspace @elastic/security-external-integrations
/x-pack/filebeat/module/ibmmq @elastic/integrations
/x-pack/filebeat/module/imperva @elastic/security-external-integrations
/x-pack/filebeat/module/infoblox @elastic/security-external-integrations
/x-pack/filebeat/module/iptables @elastic/security-external-integrations
/x-pack/filebeat/module/juniper @elastic/security-external-integrations
/x-pack/filebeat/module/microsoft @elastic/security-external-integrations
/x-pack/filebeat/module/misp @elastic/security-external-integrations
/x-pack/filebeat/module/mssql @elastic/integrations
/x-pack/filebeat/module/mysqlenterprise @elastic/security-external-integrations
/x-pack/filebeat/module/netflow @elastic/security-external-integrations
/x-pack/filebeat/module/netscout @elastic/security-external-integrations
/x-pack/filebeat/module/o365 @elastic/security-external-integrations
/x-pack/filebeat/module/okta @elastic/security-external-integrations
/x-pack/filebeat/module/oracle @elastic/security-external-integrations
/x-pack/filebeat/module/panw @elastic/security-external-integrations
/x-pack/filebeat/module/proofpoint @elastic/security-external-integrations
/x-pack/filebeat/module/rabbitmq @elastic/integrations
/x-pack/filebeat/module/radware @elastic/security-external-integrations
/x-pack/filebeat/module/snort @elastic/security-external-integrations
/x-pack/filebeat/module/snyk @elastic/security-external-integrations
/x-pack/filebeat/module/sonicwall @elastic/security-external-integrations
/x-pack/filebeat/module/sophos @elastic/security-external-integrations
/x-pack/filebeat/module/squid @elastic/security-external-integrations
/x-pack/filebeat/module/suricata @elastic/security-external-integrations
/x-pack/filebeat/module/threatintel @elastic/security-external-integrations
/x-pack/filebeat/module/tomcat @elastic/security-external-integrations
/x-pack/filebeat/module/zeek @elastic/security-external-integrations
/x-pack/filebeat/module/zookeeper @elastic/integrations
/x-pack/filebeat/module/zoom @elastic/security-external-integrations
/x-pack/filebeat/module/zscaler @elastic/security-external-integrations
```

Comment (ruflin) [reply .github/CODEOWNERS:37]: Thanks you @andrewkroh ❤️

**Comment by ruflin** [comment]
I updated the list with the details provided by @andrewkroh . I also added `@elastic/security-asset-management` permissions to the beats repo as this is required for the CODEOWNERS file to work. Single users like @aleksmaus already had write access to the repo but not the team so this wasn't an issue previously.

I know I have several 👍  on the PR but I made again a big change with the list from Andrew. Would be good to get someone to double check again.

@andrewkroh I have the suspicion you used some script for the codeowners list, can you share it if you have one? In any case, we should find a way to automate this. I would be nice if we could have local CODEOWNERS file so it can be maintained locally without having to update the global one manually.

**Comment by andrewkroh** [comment]
> I have the suspicion you used some script for the codeowners list, can you share it if you have one?

There was a big bash command, but it's gone now 😞 . Basically it listed the filebeat modules, grepped each of their names in the integrations repo's codeowners file, take the first result, and put it into that list. Then I manually reviewed the list and checked the history of modules that were not found.

File: .github/CODEOWNERS
Comment (andrewkroh) [inline .github/CODEOWNERS:5]: > The list is sorted alphabetically by directory and sub directories.

That's not true given the x-pack dirs that are listed near the beginning. Maybe we should drop most of the comments and do a `sort` on the contents.

Comments like `# All changes to the different deployments must be reviewed by data plane team.` do not add much value IMO, you can see that fact codified in `/deploy/ @elastic/elastic-agent-data-plane`.

**Review by andrewkroh — APPROVED** [review]

File: .github/CODEOWNERS
Comment (ruflin) [reply .github/CODEOWNERS:5]: I unfortunately agree. When I started there were like 20 lines and it was easy to see the groups by team but then things exploded.

What I will do is merge the current PR as is but I want to follow up with automation which automatically will mean, sorting is in place and all comments will disappear.

**Pull request merged**
